### PR TITLE
[PE-3920] Switch component

### PR DESCRIPTION
**Pull request by jllr**

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/516/ for more details.

**Comment by DiegoSanjuan** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/517/ for more details.

**Comment by DiegoSanjuan** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build was ABORTED. See http://jenkins.elasticbox.com/job/chi_pull_request/518/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/519/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/522/ for more details.

File: src/website/views/components/forms/toggle-switch.pug [hunk 1]
```diff
@@ -4,88 +4,282 @@ title: Toggle Switch
...
+        </span>
+        <span class="a-switch__label">Label</span>
+      </label>
     </div>
-    <div class="m-form__item -inline -row">
-      <input type="checkbox" class="a-input -toggle -noText" id="toggle7">
-      <label for="toggle7"></label>
-      <label class="-text" for="toggle7">Outage Alerts</label>
+
+h3 Switch with State Label
```
Comment (mattnickles) [inline src/website/views/components/forms/toggle-switch.pug]: Let's remove this example for now as it's not used in production.

File: src/custom-elements/src/components/toggle-switch/toggle-switch.tsx [hunk 2]
```diff
@@ -46,15 +43,20 @@ export class ToggleSwitch {
 
   render() {
     return (
-      [
+      <label htmlFor={`${this.el.id}-control`}
+            class={`a-switch ${this.hideLabel ? '-label--hide' : ''}`}>
         <input type="checkbox"
-               class={`a-input -toggle ${this.noText ? '-noText' : ''} ${this.extraClass ? this.extraClass : ''}`}
-               id={this.toggleId} disabled={this.disabled} checked={this.checked}
+               class={`a-switch__input ${this.extraClass && this.extraClass}`}
```
Comment (mattnickles) [inline src/custom-elements/src/components/toggle-switch/toggle-switch.tsx]: This line is returning undefined

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/531/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/549/ for more details.

**Pull request merged**